Wikipedia/Code/EditorViewController.swift

```diff
@@ -248,7 +252,10 @@ final class EditorViewController: UIViewController {
             } else {
                 self.addChildEditor(wikitext: wikitextFetchResponse.wikitext, needsReadOnly: needsReadOnly, onloadSelectRange: wikitextFetchResponse.onloadSelectRange)
             }
-            
+            if !UserDefaults.standard.didShowInformationEditingMessage && !UserDefaults.standard.isDifferentErrorBannerShown {
```

Comment:
There is an additional change to the business requirements, stated here - https://phabricator.wikimedia.org/T212318#9928399.

To accomplish each, what I would do is:

> Ensure the notice only shows if someone has chosen to "Edit Introduction" from the first edit button, or "Edit Source" from the overflow menu. Currently I see the notice when I start editing a section, and I understand that the difference in ordering should not be an issue with section edits.

For this you can try adding a simple check against this view controller's `sectionID` property. If the sectionID == nil or 0, that should target from the paths in the new requirement. You can add it as a conditional here.

> The notice disappeared very quickly and I didn't have time to read all of it - is that our normal length for this type of alerts? or could it be extended?

You can update the `WMFAlertManager`'s [showWarningAlert](https://github.com/wikimedia/wikipedia-ios/blob/main/Wikipedia/Code/WMFAlertManager.swift#L59) method to allow passing in a custom duration. Then pass in 5 seconds from here. I will kick off another build and will comment on the task for our further review.

Reply:
Sure I will take a look, and let you know once I am done with changes :)

Reply:
Hello @tonisevener, Sorry for the delay!

I have pushed changes for sectionId and also fixed conflicts.

For `WMFAlertManager`'s `showWarningAlert` method change, I did modified the function but due to `@objc` notation it is not allowing me to pass extra param. 
![image](https://github.com/user-attachments/assets/bb01e40c-7430-498f-beb5-3aaba4941f28)

I also tried to create via `WMFAppViewController.m` by creating an `@interface`, but no luck. Sorry, I am not really familiar with Obj-c.

Though One thing we can do is, use `true` for `showWarningAlert`'s `sticky` param to increase the duration to 2 seconds.

Let me know what you think!

Reply:
@rajparad No problem, thanks for your patience on this. Sometimes we have to fall back to an NSNumber in order to allow optionality with Objective-C.  I pushed a commit doing this and passed in a duration of 5.0.

`sticky=true` means the banner would not automatically dismiss at all. This could have been fine but I noticed that it seems our warning messages do not switch themes (and theme controls are very easily accessed in the editor). Rather than go down the path of fixing theming in our old Objective-C toast class I decided to go the NSNumber route.

I'll send out a new build and will merge once I get final design approval. Thanks!

Reply:
Thanks for the explanation!!